### ISSUE TITLE
[enhancement] shortcut to show - hide

### ISSUE THREAD

**Issue by omega3** [fonol/anki-search-inside-add-card#28]
Would it be possible to add shortcut for this add-on to activate / deactivate it?

"Extended Card Stats During Review" has such function to show / hide its window.

https://ankiweb.net/shared/info/1008566916

The aim is to save computer resources if this add-on is not used.

This enhancement idea was mentioned already on reddit.

**Comment by fonol** [comment]
Do you experience a higher CPU usage while using this add-on? Certainly Anki uses a tad more RAM with this on, but I only get small CPU spikes when doing an actual search, and RAM usage is like 40 mb higher at max. (Admittedly I don't have as many notes as some users)

**Comment by omega3** [comment]
> Do you experience a higher CPU usage while using this add-on?

Well, not very big, but I use very old CPU E7400, so it is better for me to save resources. 

Actually, the main point is that it occupy space in edit mode, and I have old monitor 1280x1024. It is narrow. I don't need to use it every time, only occasionally, so I would prefer to be able to **hide it (not necessarily deactivate).**
So my title of this request is not precise. I changed it.

There is another great add-on that can manage windows
Browser Maximize/Hide Table/Editor
https://ankiweb.net/shared/info/1819291495

It would be great if your add-on could behave like that, I mean show / hide.

**Comment by fonol** [comment]
I did a quick update. You can download the newest version and try it out. Should be CTRL-F by default, but you can change it in the config.

**Comment by omega3** [comment]
Oh, that was quick response. 

It works. But when I press first time Ctrl+F it hides my note type fields and shows only this add-on - like this:
https://imgur.com/TG1DNbL
And only after I hit again Ctrl+F it hides add-on and shows note type fields, but again in full screen:
https://imgur.com/a/OmyDVjb
So, there is no come back to half view.

Expected behavior would be:
Press Ctrl+F to show only note type fields.
Press again Ctrl+F togo back to initial view:
https://imgur.com/r6fAdZC

Or, **alternatively** (this seems to me the best solution) you can add another shortcut, so there would be two states:

**One** shortcut would be to hide add-on and show only note type fields and when press again the same shortcut come back to the initial view:
https://imgur.com/r6fAdZC
In this state add-on is shown in half of the window.

**Second** shortcut (as it works now) could be hide note type fields and show only add-on in full screen. And switch between both full screen views of add-on and note type fields.

Pressing first shortcut after that would come back to half view.

**Comment by fonol** [comment]
That is a bug possibly related to night mode. I did a fix like an hour ago, have you updated your addon on the newest version? If yes, and the issue persists, please disable nightmode, open the dialog again, and post the results here.

**Comment by fonol** [comment]
It should work like that: 
- If you have the window in a small state (you see the "search" button), and you press CTRL-F, it switches like you just hit the "search" button. Another CTRL-F should bring you back like you hit the "back" button.

- If you have the window in a big state ("search" button is hidden), it should simply hide/show the right column.

**Comment by omega3** [comment]
I am not sure I understand correctly what do you mean by "search" button.

I disabled night-mode. 
Regardless I have:
https://imgur.com/kWv2BuO
or
https://imgur.com/5dX53zI
when I press Ctrl+F it hides add-on and shows note type fields in full screen:
https://imgur.com/qBImGEF

**Comment by fonol** [comment]
Yea but that's basically how it works for me too and how I think it should work. With "search button" I meant the button that appears if you resize the add dialog to a size small enough that the right side disappears.

**Comment by fonol** [comment]
Unrelated, some things I noticed looking at your screenshots:
- Any particular reason you are using Whoosh instead of SQLite FTS? The latter should initialize MUCH faster.
- Are the Grid/Freeze/Random buttons always so messed up in your interface? That kinda sucks
- You could click on _Styling_, and insert like -10 in _Add To Result Area Height_, so that your bottom row (predefined searches/filtering) doesn't get cut off.

**Comment by omega3** [comment]
> Any particular reason you are using Whoosh instead of SQLite FTS? The latter should initialize MUCH faster.

Now I changed it. 

> Are the Grid/Freeze/Random buttons always so messed up in your interface? That kinda sucks

Yes, always. But it is not very important for me. Perhaps this is because of my screen resolution or because I use add-on to resize text: Set Font Size
https://ankiweb.net/shared/info/651521808
and I set font size to 19.

> You could click on Styling, and insert like -10 in Add To Result Area Height, so that your bottom row (predefined searches/filtering) doesn't get cut off.

Ok. It is better now.

> With "search button" I meant the button that appears if you resize the add dialog to a size small enough that the right side disappears.

Ok. For me it behaves the same.


> Yea but that's basically how it works for me too and how I think it should work. 

Ok. It is fine for me. So, you may consider it done. 

Thank you very much.

**Comment by fonol** [comment]
> Yes, always. But it is not very important for me. Perhaps this is because of my screen resolution or because I use add-on to resize text: Set Font Size
https://ankiweb.net/shared/info/651521808
and I set font size to 19. 

I just did a quick update, maybe it will help against the font-sizes being overwritten for you, maybe not, but you can at least try it out. I'm gonna mark this as closed for now.

**Comment by omega3** [comment]
> I just did a quick update, maybe it will help against the font-sizes being overwritten for you

Didn't work. When I decreased font size it doesn't overlap each other. 

Sorry to bother you again, but consider this: 
would it be possible for this add-on (hide / show) to somehow  save its window state. Check this:
Click on Add card. Add card window opens with this add-on visible, press Ctrl+F, click Add card to open another Add window. Add-on part will be visible. 

The same in edit: edit card, press Ctrl+F to hide, close window, go to next card in review, press edit and add-on part will be visible. 

Would it be possible for this add-on to remember its state, so that it stays hidden until I press Ctrl+F again, at least until Anki cloze.

The add-on that I linked to:
Browser Maximize/Hide Table/Editor
https://ankiweb.net/shared/info/1819291495
remember its state (even after Anki restart).

That would be very convenient.
